spec/support/test_setup.rb

```diff
@@ -0,0 +1,139 @@
+class TestSetup
+  # Class that instantiates standard set of test data for specs.
+
+  class << self
+
+    def self.method_missing(method_name, *args)
+      if @@recruitment_cycles&.key?(method_name)
+        @@recruitment_cycles[method_name].call
+      elsif @@subjects&.key?(method_name)
+        @@subjects[method_name].call
+      elsif @@enrichments&.key?(method_name)
+        @@enrichments[method_name].call
+      elsif @@courses&.key?(method_name)
+        @@courses[method_name].call
+      elsif @@site_statuses&.key?(method_name)
+        @@site_statuses[method_name].call
+      elsif @@providers&.key?(method_name)
+        @@providers[method_name].call
+      elsif @@organisations&.key?(method_name)
+        @@organisations[method_name].call
+      elsif @@users&.key?(method_name)
+        @@users[method_name].call
+      else
+        super
+      end
+    end
+
+    @@recruitment_cycles = {
+      current_cycle: -> { FactoryBot.find_or_create :recruitment_cycle },
+      next_cycle: -> { FactoryBot.find_or_create :recruitment_cycle, :next }
+    }
+
+    @@users = {
```

Comment:
Would it make sense to have 'creator services' for users, orgs and providers? (similar to the `SubjectCreatorService`). This would cut down on a lot of factory creation